transport_test.go

```diff
@@ -42,3 +57,143 @@ func TestInitTransportWithCustomCA(t *testing.T) {
 		t.Error("expected root CA to be set, got nil")
 	}
 }
+
+type testHTTPHandler struct {
+	t                 *testing.T
+	gotCorrectRequest chan (bool)
+}
+
+func (h *testHTTPHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	if len(req.TLS.PeerCertificates) > 0 {
+		h.gotCorrectRequest <- true
+	} else {
+		reqDump, _ := httputil.DumpRequest(req, false)
+		h.t.Logf("got request without client certificates:\n%s", reqDump)
+		h.t.Logf("TLS config: %#v", req.TLS)
+
+		h.gotCorrectRequest <- false
```

Comment:
Couldn't we signal the result by using HTTP Status Codes instead and put the logs into the body?

Reply:
You still need to wait for the server to acknowledge that it got the request and processed it.

We could send the error response in a body.

Reply:
I was so stuck on my solution that I almost forgot how normal client-server communication works 

I'll rework this, then.